tests fail on OSX El Capitan 10.11.5

tox

```
======================================== ERRORS =========================================
________________ ERROR at setup of test_direct_transfer_callback[2-mock] ________________

request = <SubRequest 'raiden_network' for <Function 'test_direct_transfer_callback[2-mock]'>>
assets_addresses = ['F\x0c?/xIP\\Q{\t*Be~h=>>M'], channels_per_node = 1, deposit = 200
settle_timeout = 600
blockchain_services = BlockchainServices(deploy_service=<raiden.tests.utils.mock_client.BlockChainSe... <raiden.tests.utils.mock_client.BlockChainServiceMock object at 0x107203190>])
transport_class = <class 'raiden.network.transport.UDPTransport'>, send_ping_time = 0
max_unresponsive_time = 0, cached_genesis = None

    @pytest.fixture
    def raiden_network(
            request,
            assets_addresses,
            channels_per_node,
            deposit,
            settle_timeout,
            blockchain_services,
            transport_class,
            send_ping_time,
            max_unresponsive_time,
            cached_genesis):

        verbosity = request.config.option.verbose

        raiden_apps = create_apps(
            blockchain_services.blockchain_services,
            transport_class,
            verbosity,
            send_ping_time,
>           max_unresponsive_time
        )

raiden/tests/fixtures/raiden_network.py:98:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
raiden/tests/utils/network.py:292: in create_apps
    host=host
raiden/tests/utils/network.py:61: in create_app
    transport_class,
raiden/app.py:30: in __init__
    self.transport = transport_class(config['host'], config['port'])
raiden/network/transport.py:74: in __init__
    self.server.start()
.tox/py27/lib/python2.7/site-packages/gevent/baseserver.py:291: in start
    self.init_socket()
.tox/py27/lib/python2.7/site-packages/gevent/server.py:142: in init_socket
    self.socket = self.get_listener(self.address, self.family)
.tox/py27/lib/python2.7/site-packages/gevent/server.py:152: in get_listener
    return _udp_socket(address, reuse_addr=self.reuse_addr, family=family)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

address = ('127.0.0.10', 40001), backlog = 50, reuse_addr = 1, family = 2

    def _udp_socket(address, backlog=50, reuse_addr=None, family=_socket.AF_INET):
        # we want gevent.socket.socket here
        sock = socket(family=family, type=_socket.SOCK_DGRAM)
        if reuse_addr is not None:
            sock.setsockopt(_socket.SOL_SOCKET, _socket.SO_REUSEADDR, reuse_addr)
        try:
>           sock.bind(address)
E           error: [Errno 49] Can't assign requested address: ('127.0.0.10', 40001)

.tox/py27/lib/python2.7/site-packages/gevent/server.py:194: error
!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!
```

(same tox)

```
======================================== ERRORS =========================================
________________ ERROR at setup of test_direct_transfer_callback[2-mock] ________________

request = <SubRequest 'raiden_network' for <Function 'test_direct_transfer_callback[2-mock]'>>
assets_addresses = ['x"\tZbZIPOU1DKF\x0b)O($)'], channels_per_node = 1, deposit = 200
settle_timeout = 600
blockchain_services = BlockchainServices(deploy_service=<raiden.tests.utils.mock_client.BlockChainSe... <raiden.tests.utils.mock_client.BlockChainServiceMock object at 0x104338610>])
transport_class = <class 'raiden.network.transport.UDPTransport'>, send_ping_time = 0
max_unresponsive_time = 0, cached_genesis = None

    @pytest.fixture
    def raiden_network(
            request,
            assets_addresses,
            channels_per_node,
            deposit,
            settle_timeout,
            blockchain_services,
            transport_class,
            send_ping_time,
            max_unresponsive_time,
            cached_genesis):

        verbosity = request.config.option.verbose

        raiden_apps = create_apps(
            blockchain_services.blockchain_services,
 node010 -> node@0.10
gcc49 -> gcc@4.9                             libmongoclient -> mongo-cxx-driver           node012 -> node@0.12
==> Deleted Formulae
cpp-netlib                                   homebrew/versions/gcc46                      homebrew/versions/libgee08                   homebrew/versions/postgresql93
homebrew/versions/allegro4                   homebrew/versions/gcc47                      homebrew/versions/libidn2                    homebrew/versions/povray36
homebrew/versions/android-ndk-r10e           homebrew/versions/gcc48                      homebrew/versions/libmongoclient-legacy      homebrew/versions/ppl011
homebrew/versions/android-ndk-r11c           homebrew/versions/gcc49                      homebrew/versions/libmpc08                   homebrew/versions/ppl10
homebrew/versions/ansible19                  homebrew/versions/gcc5                       homebrew/versions/libotr4                    homebrew/versions/qt52
homebrew/versions/ansible20                  homebrew/versions/gcc6                       homebrew/versions/libuv0                     homebrew/versions/qt55
homebrew/versions/antlr2                     homebrew/versions/gdal111                    homebrew/versions/libxc21                    homebrew/versions/rabbitmq30
homebrew/versions/antlr3                     homebrew/versions/gecode373                  homebrew/versions/llvm-gcc28                 homebrew/versions/raptor1
homebrew/versions/apache-spark15             homebrew/versions/gegl02                     homebrew/versions/llvm33                     homebrew/versions/riak132
homebrew/versions/apache-spark16             homebrew/versions/git-tf-2.0.2               homebrew/versions/llvm34                     homebrew/versions/ruby192
address = ('127.0.0.10', 40001), backlog = 50, reuse_addr = 1, family = 2

    def _udp_socket(address, backlog=50, reuse_addr=None, family=_socket.AF_INET):
        # we want gevent.socket.socket here
        sock = socket(family=family, type=_socket.SOCK_DGRAM)
        if reuse_addr is not None:
            sock.setsockopt(_socket.SOL_SOCKET, _socket.SO_REUSEADDR, reuse_addr)
        try:
>           sock.bind(address)
E           error: [Errno 49] Can't assign requested address: ('127.0.0.10', 40001)

.tox/coverage/lib/python2.7/site-packages/gevent/server.py:194: error
!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/raiden/tests/conftest.py
+++ b/raiden/tests/conftest.py
@@ -140,6 +140,13 @@ def pytest_addoption(parser):
     )
 
     parser.addoption(
+        '--initial-port',
+        type=int,
+        default=29870,
+        help='Base port number used to avoid conflicts while running parallel tests.',
+    )
+
+    parser.addoption(
         '--log-config',
         default=None,
     )

--- a/raiden/tests/fixtures/blockchain.py
+++ b/raiden/tests/fixtures/blockchain.py
@@ -12,6 +12,7 @@ from pyethapp.rpc_client import JSONRPCClient
 from pyethapp.jsonrpc import address_decoder, address_encoder, default_gasprice
 
 from raiden.utils import privatekey_to_address, get_contract_path, safe_lstrip_hex
+from raiden.network.transport import DummyTransport
 from raiden.tests.fixtures.tester import tester_state
 from raiden.tests.utils.blockchain import GENESIS_STUB, DEFAULT_BALANCE_BIN
 from raiden.tests.utils.mock_client import BlockChainServiceMock, MOCK_REGISTRY_ADDRESS
@@ -111,7 +112,7 @@ def cached_genesis(request, blockchain_type):
     raiden_apps = create_apps(
         blockchain_services,
         request.getfixturevalue('raiden_udp_ports'),
-        request.getfixturevalue('transport_class'),
+        DummyTransport,  # Do not use a UDP server to avoid port reuse in MacOSX
         request.config.option.verbose,
         request.getfixturevalue('send_ping_time'),
         request.getfixturevalue('max_unresponsive_time'),

--- a/raiden/tests/fixtures/variables.py
+++ b/raiden/tests/fixtures/variables.py
@@ -166,9 +166,9 @@ def blockchain_private_keys(blockchain_number_of_nodes, blockchain_key_seed):
 
 
 @pytest.fixture(scope='session')
-def port_generator():
+def port_generator(request):
     """ count generator used to get a unique port number. """
-    return count(29870)
+    return count(request.config.option.initial_port)
 
 
 @pytest.fixture

--- a/tox.ini
+++ b/tox.ini
@@ -23,10 +23,10 @@ deps =
 commands = coverage run --append -m py.test --blockchain-type=tester --exitfirst {posargs:./raiden/}
 
 [testenv:smart_contracts]
-commands = coverage run --append -m py.test --blockchain-type=tester --exitfirst ./raiden/tests/smart_contracts {posargs}
+commands = coverage run --append -m py.test --initial-port=2000 --blockchain-type=tester --exitfirst ./raiden/tests/smart_contracts {posargs}
 
 [testenv:integration]
-commands = coverage run --append -m py.test --blockchain-type=tester --exitfirst ./raiden/tests/integration {posargs}
+commands = coverage run --append -m py.test --initial-port=3000 --blockchain-type=tester --exitfirst ./raiden/tests/integration {posargs}
 
 [testenv:unit]
-commands = coverage run --append -m py.test --blockchain-type=tester --exitfirst ./raiden/tests/unit {posargs}
+commands = coverage run --append -m py.test --initial-port=4000 --blockchain-type=tester --exitfirst ./raiden/tests/unit {posargs}

